File: JAVA/lecture5/GO/src/Point.java
public class Point extends GraphicalObject
{
    public Point(int color, int x, int y) {
        super(color);
        setX(x);
        setY(y);
    }

    public Point(Point point)
    {
        super(point.getColor());
        setX(point.getX());
        setY(point.getY());
    }

    public final int getX()
    {
        return x;
    }

    public final void setX(int x)
    {
        this.x = x;
    }

    public final int getY()
    {
        return y;
    }

    public final void setY(int y)
    {
        this.y = y;
    }

    public Point kopie()
    {
        return new Point(this);
    }

    @Override
    public String toString() {
        return super.toString() + "(" + x + ", " + y + ")";
    }

    @Override
    public void draw()
    {
//        System.out.println(this);
        super.draw();
        System.out.print("(" + x + ", " + y + ")");
    }

    private int x;
    private int y;
}

File: CIS/lecture4/ctvrty/Point.cs
namespace Ctvrty
{
  internal class Point : GraphicalObject
  {
    public int X { get; set; }
    public int Y { get; set; }

    public Point(int x, int y, int color) : base(color)
    {
      X = x;
      Y = y;
    }

    public override void Draw()
    {
      base.Draw();
      Console.Write(" ({0}, {1})", X, Y);
    }

    public override Point CopyOfObject()
    {
      return new Point(X, Y, Color);
    }
  }
}
